### PR TITLE
fix: Tenderly simulation initCode split, 7702 sentinel forms, and bigint precision

### PR DESCRIPTION
**Pull request by sherifahmed990**
Fixes in `utilsTenderly` that made simulations wrong or lossy.

  - **Every v0.6 deployment simulation was failing**: the v0.6 `initCode` was split at 22 chars instead of 42, truncating the factory address to 10 bytes and leaking its second half
  into an unprefixed `factoryData`.
  - **EIP-7702 sentinel only matched the short form**: the check matched `0x7702` but not the 20-byte right-padded form accepted by EntryPoint v0.8, which fell through — skipping the delegation code override, or triggering a bogus deployment simulation in the callData path.
  - **Precision loss above 2^53**: `callTenderlySimulateBundle` typed `value`/`gasPrice` as `number`, silently rounding large wei amounts. `bigint` is now accepted and serialized as a decimal string.
  - **Caller's `stateOverrides` mutated in place**: the `stateDiff` → `storage` rename rewrote the caller's object; the conversion now works on a copy.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved EIP-7702 delegation simulation to correctly support both compact and padded factory sentinel forms.
  - Fixed factory address and factory-data extraction when parsing combined init-code in call-data simulations.
  - Prevented precision loss by serializing large gas price and value inputs safely.
  - Enhanced state override processing with stricter validation and non-mutating behavior, including correct handling of storage/state-diff formats.
- **Tests**
  - Added regression coverage for EIP-7702 simulation behaviors and override merging/non-mutation.

### CODE REVIEW

File: src/utilsTenderly.ts [hunk 1]
```diff
@@ -247,7 +263,9 @@ export async function simulateUserOperationWithTenderly(
 	// override so the simulation passes.
 	if (
 		!isV6UserOperation &&
-		(userOperation as UserOperationV7 | UserOperationV8 | UserOperationV9).factory === "0x7702"
+		isEip7702FactorySentinel(
```
Comment (Sednaoui) [inline src/utilsTenderly.ts:279]: The expanded padded-marker detection is valid, but the short-marker case is still malformed when `factoryData` is non-empty. The `initCode` construction above concatenates the data immediately after `0x7702`, producing e.g. `0x77021234`; EntryPoint requires `0x7702` right-padded to 20 bytes before the initialization data. A live Tenderly run decoded exactly `0x77021234` and reverted through the ordinary deployment path with `AA10 sender already constructed`. Please normalize the short marker during `initCode` construction and add a regression test for non-empty `factoryData`.

File: src/utilsTenderly.ts [hunk 2]
```diff
@@ -488,7 +507,7 @@ export async function simulateUserOperationCallDataWithTenderly(
 		// EIP-7702 userOps use factory:"0x7702" as a sentinel with
 		// factoryData:null. This doesn't represent an actual factory
 		// deployment, so normalize to null.
-		if (factory === "0x7702") {
+		if (isEip7702FactorySentinel(factory)) {
```
Comment (Sednaoui) [inline src/utilsTenderly.ts:525]: This avoids treating the padded marker as a factory, but the direct simulation is still semantically incomplete: it neither installs the sender's EIP-7702 delegation-code override nor preserves non-empty `factoryData` as the sender initialization call. In live Tenderly simulation `001431d4-9e83-409d-b667-49cafe163d16`, this path returned `status=true` while executing no internal transfer; adding the missing code override made the same call execute successfully (`c727864a-f42e-4e0e-8228-371482f24fa2`). Please mirror the full simulation's delegation override here and, when `factoryData` is non-empty, enqueue the SenderCreator-to-sender initialization before the call-data transaction.

**Review by Sednaoui — CHANGES_REQUESTED** [review]
Requesting changes based on live Tenderly verification.

What is working:
- The padded EIP-7702 marker in the full `handleOps` path is valid. Simulation `07bb5508-667b-4deb-8580-c4af19bfd61e` completed with `status=true`, included the sender code override, and executed the expected 1-wei internal transfer.
- The v0.6 split, bigint serialization, and non-mutating state-override changes also behaved correctly in isolated request probes.

Two EIP-7702 cases still need correction:
1. The short marker with non-empty `factoryData` is packed as `0x7702<factoryData>` rather than a 20-byte right-padded marker followed by the data.
2. The direct call-data path removes the sentinel but does not install the delegation override and discards initialization data. In simulation `001431d4-9e83-409d-b667-49cafe163d16`, Tenderly returned `status=true` but no internal transfer executed. The same call with the missing delegation override added (`c727864a-f42e-4e0e-8228-371482f24fa2`) executed successfully.

Please add regression coverage for short/padded markers, fresh delegation, and non-empty `factoryData`.

File: src/utilsTenderly.ts (hunk 2 above)
Comment (sherifahmed990) [reply src/utilsTenderly.ts:525]: fixed

**Review by Sednaoui — APPROVED** [review]
